src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -997,6 +1175,52 @@ export class SavedObjectsRepository {
       }
     }
 
+    if (!disableExtensions && this._spacesExtension) {
+      try {
+        namespaces = await this._spacesExtension.getSearchableNamespaces(options.namespaces);
+      } catch (err) {
+        if (Boom.isBoom(err) && err.output.payload.statusCode === 403) {
+          // The user is not authorized to access any space, return an empty response.
+          return SavedObjectsUtils.createEmptyFindResponse<T, A>(options);
+        }
+        throw err;
+      }
+      if (namespaces.length === 0) {
+        // The user is authorized to access *at least one space*, but not any of the spaces they requested; return an empty response.
+        return SavedObjectsUtils.createEmptyFindResponse<T, A>(options);
+      }
+    }
+
+    // We have to first do a "pre-authorization" check so that we can construct the search DSL accordingly
+    const spacesToPreauthorize = new Set(namespaces);
+    let typeToNamespacesMap: Map<string, string[]> | undefined;
+    let preAuthorizationResult: CheckAuthorizationResult<'find'> | undefined;
+    if (!disableExtensions && this._securityExtension) {
+      preAuthorizationResult = await this._securityExtension?.checkAuthorization({
+        types: new Set(types),
+        spaces: spacesToPreauthorize,
+        actions: ['find'],
+      });
+      if (preAuthorizationResult.status === 'unauthorized') {
+        // If the user is unauthorized to find *anything* they requested, return an empty response
+        this._securityExtension.addAuditEvent({
+          action: AuditAction.FIND,
+          error: new Error('unauthorized'), // TODO: improve this error message?
+          // TODO: include object type(s) that were requested?
```

Comment:
Should the AuditEvent interface be modified to include requested types and requested spaces? Or should we just add these to the error message?

Reply:
The audit logging that existed in the security wrapper for this scenario doesn't include the types, so I think we can leave it out of scope for this PR, but open a separate issue for us to address as a followup item. I agree it would be worthwhile, but I'm also keen not to add unnecessary scope to this PR.

Reply:
Ok. I at least improved the messages for the audits in question. I'll open an issue for augmenting the SO audits with additional parameters.